### DISCUSSION TITLE
Funktion closestTo

### DISCUSSION THREAD

**Discussion by therealchrismcfly** [neuefische/questions#134] [Web] [answered]
Guten Morgen,

ich habe folgende Frage: 

Ich möchte, dass auf der Startseite im Footer der einzelnen Übungen immer die Werte für Sets, Reps und Weight angezeigt werden, die am nächsten am aktuellen Datum dran sind. Dazu habe ich die Funktion closestTo benutzt: 

https://date-fns.org/v2.29.2/docs/closestTo

Die Funktion closestTo funktioniert, das hab ich mit console.log getestet. Nun fehlt mir aber der Bogen dazu, wie ich zu den einzelnen Datumseinträgen mit Sets, Weights, Reps kommen für die jeweiligen Exercises. 

![Bildschirmfoto 2022-09-04 um 12 06 50](https://user-images.githubusercontent.com/107792731/188308224-7a3a1ac9-9340-441d-8eb2-f3e7062d19b5.png)

Hier der Link zum aktuellen Repo: https://github.com/therealchrismcfly/capstone-project/tree/styling 

Danke schonmal für eure Hilfe! :-)

Chris

**Comment by alexclaes** [comment, marked as answer]
Ich würde vorschlagen, das Array `correspondingExercises` einfach mit `.sort()` absteigend zu sortieren und dann den ersten Eintrag zu betrachten. Das sollte einfacher sein, um das neuste Workout zu ermitteln.

Wenn du mit `closestTo()` arbeite möchtest, musst du erst wieder das passende Element aus dem `correspondingExercises` anhand des Datums heraussuchen.

z.B.

```js
const lastWorkout = correspondingExercises.find(workout => isEqual(workout.date, latestProgress));
```

https://date-fns.org/v2.29.2/docs/isEqual

**Reply by therealchrismcfly** [reply]
Danke Alex, das ist ein guter Tipp. Wo es bei der .sort() Sache schon hing ist die folgende Fehlermeldung beim mappen: 

TypeError: Cannot read properties of undefined (reading 'map')

![Bildschirmfoto 2022-09-05 um 14 32 30](https://user-images.githubusercontent.com/107792731/188450172-ec6f9ef2-410f-4d6f-b956-9c5f4918fc47.png)

**Reply by alexclaes** [reply]
Wie sieht denn der Code aus, indem du `.sort()` anwendets? Wo wird der Variable `lastWorkouts` ein Wert zugewiesen? 

Wenn `lastWorkouts` an dieser Stelle den Wert `undefined` hat,  wird der Fehler dort stecken.
